api/wikipediaSource.ts

```diff
@@ -0,0 +1,80 @@
+const BASE_URL: string = "https://en.wikipedia.org"
+const ENDPOINT: string = "/w/api.php?"
+
+
+/**
+ * Fetch and return the introduction as plain text of the given Wikipedia page.
+ * Uses the MediaWiki Action API.
+ * @param pageTitle the title of the Wikipedia page, must be first-capitalized
+ * words (except for name particles) separated by '_', for instance "Leonardo_da_Vinci"
+ * @throws Error if no page corresponding to the given title is found
+ */
+async function fetchIntro(pageTitle: string): Promise<string> {
```

Comment:
You don't need `async` here, since you are using `fetch().then()...` and not `await fetch()`
```suggestion
function fetchIntro(pageTitle: string): Promise<string> {
```

Reply:
My IDE tells me it should be converted to an async function. It is actually rendering a `Promise`, which means the actual result is asynchronous.